imblearn/utils/_validation.py

```diff
@@ -93,7 +101,7 @@ def check_neighbors_object(nn_name, nn_object, additional_neighbor=0):
     """
     if isinstance(nn_object, Integral):
         return NearestNeighbors(n_neighbors=nn_object + additional_neighbor)
-    elif isinstance(nn_object, KNeighborsMixin):
+    elif _is_neighbors_object(nn_object):
         return clone(nn_object)
     else:
         raise_isinstance_error(nn_name, [int, KNeighborsMixin], nn_object)
```

Comment:
I think that we should as well change the error message since we don't strictly require to be a `KNeighborsMixin` but instead to expose both `kneighbors` and `kneighbors_graph`.

Reply:
I have been debating between two implementations here (my two latest commits [[1](https://github.com/scikit-learn-contrib/imbalanced-learn/pull/858/commits/f10405783c7463fe52981d9651072a48a784aa12) [2](https://github.com/scikit-learn-contrib/imbalanced-learn/pull/858/commits/b82e4d95279b77f310026b24d8429772b05a531a)]).

[1] uses sklearn.base.clone to verify that the nn_object is an sklearn-like estimator that can be cloned.  This implementation is more consistent with how the library checks the integrity of other estimators - such as the KMeans Estimator check in [`KMeansSmote`](https://github.com/scikit-learn-contrib/imbalanced-learn/blob/f177b05256fa3e4e556c1a8be0283b721b1bcd86/imblearn/over_sampling/_smote/cluster.py#L148-L158), but it does not protect users from the mistake of inputting other types of estimators.

[2] raises a TypeError if the nn_object is neither an integer, nor exposes both `kneighbors` and `kneighbors_graph`;  thus, it protects users from this potential mistake.

Do you prefer one over the other?